data-model/src/main/java/io/micronaut/data/model/query/builder/sql/SqlQueryBuilder.java

```diff
@@ -787,14 +787,19 @@ protected void selectAllColumns(QueryState queryState, StringBuilder queryBuffer
                         } else {
                             columnName = asPath(propertyAssociations, prop);
                         }
+                        String columnAlias = getColumnAlias(prop);
+
                         queryBuffer
                                 .append(aliasName)
                                 .append(DOT)
                                 .append(queryState.shouldEscape() ? quote(columnName) : columnName)
-                                .append(AS_CLAUSE)
-                                .append(joinPathAlias)
-                                .append(columnName)
-                                .append(COMMA);
+                                .append(AS_CLAUSE);
+                        if (StringUtils.isNotEmpty(columnAlias)) {
```

Comment:
Not sure how would aliases work in the context of JPA (computePropertyPaths() == false)

Reply:
I will have to check that path, needed to add changes for DTO as well.

Reply:
Added test with JPA specification, works ok - if that's what you meant here ^